sql/src/main/java/io/crate/analyze/symbol/Symbols.java

```diff
@@ -168,4 +166,38 @@ public Boolean visitReference(Reference symbol, ColumnIdent context) {
             return context.equals(symbol.ident().columnIdent());
         }
     }
+
+    private static class RCReplaceCtx {
+        private final Predicate<? super RelationColumn> predicate;
+        private final com.google.common.base.Function<? super RelationColumn, ? extends Symbol> replaceFunction;
+
+        RCReplaceCtx(Predicate<? super RelationColumn> predicate,
+                     com.google.common.base.Function<? super RelationColumn, ? extends Symbol> replaceFunction) {
+            this.predicate = predicate;
+            this.replaceFunction = replaceFunction;
+        }
+    }
+
+    private static class RCReplaceVisitor extends ReplacingSymbolVisitor<RCReplaceCtx> {
+
+        public final static RCReplaceVisitor INSTANCE = new RCReplaceVisitor();
+
+        private RCReplaceVisitor() {
+            super(ReplaceMode.COPY);
+        }
+
+        @Override
+        public Symbol visitRelationColumn(RelationColumn relationColumn, RCReplaceCtx ctx) {
+            if (ctx.predicate.apply(relationColumn)) {
+                return ctx.replaceFunction.apply(relationColumn);
+            }
+            return relationColumn;
+        }
+    }
+
+    public static Symbol replaceRC(Symbol symbol,
+                                   Predicate<? super RelationColumn> predicate,
```

Comment:
Might make sense to remove the predicate. The function could do the filtering as well - would remove the need for a separate RCReplaceCtx